actions/pr-slack-alert/action.yml

```diff
@@ -0,0 +1,120 @@
+name: pr-slack-alert
+description:
+  "Send Slack alert to a specific user group in a channel from Github PRs"
+
+inputs:
+  # general inputs
+  checkout-repo:
+    description: "enable git checkout repo"
+    required: false
+    default: "true"
+  checkout-repo-fetch-depth:
+    description: "number of commits to fetch"
+    required: false
+    default: "0"
+  # slack inputs
+  slack-channel-id:
+    description: "slack channel id to send the alert to"
+    required: true
+  slack-user-group-id:
+    description: "slack user group id to ping with the alert"
+    required: true
+  slack-bot-token:
+    description:
+      "slack bot api token - the token requires `chat:write` permission"
+    required: true
+  # github inputs
+  ignore-pr-creators:
+    description: "comma-separated list of github usernames to skip alerts for"
+    required: false
+    default: "github-actions[bot],dependabot[bot],renovate[bot]"
+  match-reviewer-team:
+    description:
+      "will only send alert if the team name is included in the reviewers list"
+    required: false
+    default: ""
+  pr-event-type:
+    description:
+      "filter to trigger on specific PR event type - if set to empty it will
+      trigger on every type"
+    required: false
+    default: ""
+  # grafana inputs (optional)
+  metrics-job-name:
+    description: "grafana metrics job name"
+    required: false
+  metrics-id:
+    description:
+      "grafana metrics id, used for continuity of metrics during job name
+      changes  - required if metrics-job-name is passed"
+    required: false
+  gc-host:
+    description: "grafana hostname - required if metrics-job-name is passed"
+    required: false
+  gc-basic-auth:
+    description: "grafana basic auth - required if metrics-job-name is passed"
+    required: false
+  gc-org-id:
+    description:
+      "grafana org/tenant id - required if metrics-job-name is passed"
+    required: false
+
+runs:
+  using: composite
+  steps:
+    - name: Checkout repo
+      if: inputs.checkout-repo == 'true'
+      uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+      with:
+        fetch-depth: ${{ inputs.checkout-repo-fetch-depth }}
```

Comment:
I don't know why this is in the boiler-plate. Is there a reason why we should include this here, and not let the action's users checkout the repo before calling this?

Treat this comment as non-blocking by the way, can be merged with this, just curious at your take on this.

Reply:
yeah very true - I think we've just stuck with this pattern but never really questioned why lol

Reply:
removed it :)